### PR TITLE
Convert views from using site_id to site_slug

### PR DESCRIPTION
**Pull request by lo9key**
fixes issue #133

## Changes in this PR.

- [X] Convert all views from using site_id to site_slug, where appropriate.
- [X] Fix all test issues caused by the change
- [X] Passes flake8

## Testing this PR.

1 - You'll probably need to migrate your database to apply any changes to models.py:
```
$ docker-compose run web bash
# python streamwebs_frontend/manage.py makemigrations
# python streamwebs_frontend/manage.py migrate
```
2 - Run tests using `# ./runtests`
3 - Check flake8 using `# flake8`

### Expected Output.

```
# flake8
./docs/source/conf.py:284:1: E265 block comment should start with '# '
( lots of these)
./streamwebs_frontend/streamwebs/migrations/0001_initial.py:27:80: E501 line too long (114 > 79 characters)
( lots of these)
```
You should not see any flake8 warnings except for `docs/source/conf.py` and `streamwebs_frontend/streamwebs/migrations/*`.

```
# ./runtests.sh 
No changes detected
Operations to perform:
  Apply all migrations: admin, contenttypes, streamwebs, auth, sessions
Running migrations:
  No migrations to apply.
Creating test database for alias 'default'...
...................................................................................Invalid login details: notjohn, badpassword
.Invalid login details: notjohn, johnpassword
.Invalid login details: john, badpassword
.............<ul class="errorlist"><li>password<ul class="errorlist"><li>Passwords do not match</li></ul></li></ul> 
.<ul class="errorlist"><li>username<ul class="errorlist"><li>This field is required.</li></ul></li><li>password_check<ul class="errorlist"><li>This field is required.</li></ul></li><li>password<ul class="errorlist"><li>This field is required.</li></ul></li><li>email<ul class="errorlist"><li>This field is required.</li></ul></li></ul> <ul class="errorlist"><li>captcha<ul class="errorlist"><li>This field is required.</li></ul></li><li>school<ul class="errorlist"><li>This field is required.</li></ul></li><li>birthdate<ul class="errorlist"><li>This field is required.</li></ul></li></ul>
. <ul class="errorlist"><li>captcha<ul class="errorlist"><li>Incorrect, please try again.</li></ul></li></ul>
. <ul class="errorlist"><li>captcha<ul class="errorlist"><li>Incorrect, please try again.</li></ul></li></ul>
.....
----------------------------------------------------------------------
Ran 106 tests in 1.752s

OK
Destroying test database for alias 'default'...
```

@osuosl/devs

### CODE REVIEW

File: streamwebs_frontend/streamwebs/models.py [hunk 1]
```diff
@@ -29,10 +29,10 @@ class Site(models.Model):
     location as a pair of latitudinal/longitudinal coordinates and an optional
     text description of entry.
     """
-    site_name = models.CharField(max_length=250, blank=True)
+    site_name = models.CharField(max_length=250, blank=False)
     site_type = models.CharField(max_length=250, blank=True)
     description = models.TextField(blank=True)
-    site_slug = models.SlugField(blank=True)
+    site_slug = models.SlugField(unique=True, blank=True, editable=False)
```
Comment (leian7) [inline streamwebs_frontend/streamwebs/models.py]: When would we want site_slug to be blank?

File: streamwebs_frontend/streamwebs/tests/models/test_site_model.py [hunk 2]
```diff
@@ -51,3 +51,6 @@ def test_optional_fields(self):
         for field in self.optional_fields:
             self.assertEqual(
                 Site._meta.get_field(field).blank, True)
+
+    def test_site_slug_is_unique(self):
+        self.site = Site.test_objects.create_site('test site', 'site type')
```
Comment (leian7) [inline streamwebs_frontend/streamwebs/tests/models/test_site_model.py]: Asserts?

**Comment by lo9key** [comment]
@leian7 Great feedback, thanks!  I added tests to ensure site_slug is unique, and not too long.

**Comment by LyonesGamer** [comment]
+1

**Comment by leian7** [comment]
Very cool. :+1:

**Pull request merged**